var/spack/repos/builtin/packages/netcdf-fortran/package.py

```diff
@@ -0,0 +1,18 @@
+from spack import *
+
+class NetcdfFortran(Package):
+    """Fortran interface for NetCDF4"""
+
+    homepage = "http://www.unidata.ucar.edu/downloads/netcdf/netcdf-cxx/index.jsp"
+    url      = "http://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-fortran-4.4.3.tar.gz"
+
+    version('4.4.3', 'bfd4ae23a34635b273d3eb0d91cbde9e')
+
+    variant('mpi', default=True, description='Enables MPI parallelism')
```

Comment:
No, in the concretization step, the `spec` DAG is normalized so that there's only one node for each package.  To do that, all the constraints are merged.  So, if some package A has the above dependencies, and some other package B just depends on `netcdf`, the merge will succeed, and`+mpi` will be added to the common `netcdf` spec.  If B depends on `netcdf~mpi` and it somehow ends up as a dep in the same `spec` as A, you'll get an error (probably `UnsatisfiableVariantSpecError`) telling you about the constraint violation.